llvm/lib/Transforms/Utils/SSAUpdaterBulk.cpp

```diff
@@ -222,3 +223,38 @@ void SSAUpdaterBulk::RewriteAllUses(DominatorTree *DT,
     }
   }
 }
+
+// Perform a single pass of simplification over the worklist of PHIs.
+static void simplifyPass(MutableArrayRef<PHINode *> Worklist,
+                         const DataLayout &DL) {
+  for (PHINode *&PHI : Worklist) {
+    if (Value *Simplified = simplifyInstruction(PHI, DL)) {
+      PHI->replaceAllUsesWith(Simplified);
```

Comment:
Can't RewriteAllUses do this as it sees the phis in the first place?

Reply:
I thought about doing this but it complicates things: this would require to use tracking VH in the BBInfo structure, I'm not sure it worth it.